### PR TITLE
feat: pagination for index cases and events lists

### PR DESCRIPTION
**Pull request by faversente**
BREAKING CHANGE: 

This PR introduces API driven pagination (+ sort, search and filter) for "getAll" functionality in index cases and events lists.

### CODE REVIEW

**Comment by faversente** [comment]
All proposed and discussed changes should be implemented now.
@bianucci please re-review.

File: iris-client-bff/src/main/java/iris/client_bff/cases/web/IndexCaseController.java [hunk 1]
```diff
@@ -35,11 +27,18 @@
...
-
-		return indexCaseService.findAll().stream()
-				.map(IndexCaseMapper::map)
-				.collect(Collectors.toList());
+	public Page<IndexCaseDTO> getAll(@RequestParam(required = false) Status status,
+			@RequestParam(required = false) String search, Pageable pageable) {
+		if (status != null && search != null) {
+			return indexCaseService.findByStatusAndSearchByRefIdOrName(status, search, pageable).map(IndexCaseMapper::map);
+		}
+		else if (search != null) {
```
Comment (bianucci) [inline iris-client-bff/src/main/java/iris/client_bff/cases/web/IndexCaseController.java:35]: Lets use `StringUtils.isNotEmpty`

File: iris-client-bff/src/main/java/iris/client_bff/events/web/EventDataRequestController.java [hunk 2]
```diff
@@ -64,12 +58,19 @@
...
-		}).collect(Collectors.toList());
-		var res = ExistingDataRequestClientWithLocationList.builder().dataRequests(response).build();
-		return ResponseEntity.of(Optional.of(res));
+	public Page<ExistingDataRequestClientWithLocation> getDataRequests(
+			@RequestParam(required = false) Status status,
+			@RequestParam(required = false) String search, Pageable pageable) {
+		if (status != null && search != null) {
+			return dataRequestService.findByStatusAndSearchByRefIdOrName(status, search, pageable)
+					.map(eventMapperFunction);
+		} else if (search != null) {
```
Comment (bianucci) [inline iris-client-bff/src/main/java/iris/client_bff/events/web/EventDataRequestController.java:67]: Same here: lets use StringUtils

File: iris-client-bff/src/test/java/iris/client_bff/events/web/EventDataRequestControllerIntegrationTest.java [hunk 3]
```diff
@@ -40,43 +51,126 @@
...
+				.andReturn();
 	}
 
 	@Test
 	@WithMockUser()
 	public void updateDataRequestByCode() throws Exception {
 		String REFID = "refId";
-		var dataRequest = Mockito.spy(new EventDataRequest(REFID, "name", Instant.now(), Instant.now(), "", "requestDetails", "", new Location()));
+		var dataRequest = Mockito.spy(
+				new EventDataRequest(
```
Comment (bianucci) [inline iris-client-bff/src/test/java/iris/client_bff/events/web/EventDataRequestControllerIntegrationTest.java:155]: Let's use the Builder API.

File: iris-client-fe/src/api/common.ts [hunk 4]
```diff
@@ -29,110 +30,150 @@ export const DUMMY_BASE_URL = 'https://example.com'
...
+  status?: DataRequestStatus | null,
+  search?: string | null,
+  sortOrderDesc?: boolean,
 }
+
+/**
+ *
+ * @export
+ */
+export interface DataPage<T> {
```
Comment (bianucci) [inline iris-client-fe/src/api/common.ts:158]: I think this type should not be defined here. Consumers should use `PageIndexCase` or `PageEvent`.

Lets discuss this directly.

File: iris-client-fe/src/utils/misc.ts [hunk 5]
```diff
@@ -1,7 +1,42 @@
...
+import { Route } from "vue-router";
+import { DataRequestStatus } from "@/api";
 
 export const join = (arr: Array<unknown>, separator: string): string => {
   return flattenDeep(arr)
     .filter((n) => n)
     .join(separator);
 };
+
+export function getStringParamFromRouteWithOptionalFallback(
```
Comment (bianucci) [inline iris-client-fe/src/utils/misc.ts:11]: Lets introduce a `Pagination.ts` utils module for this code.

File: iris-client-fe/src/constants/QueryParams.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export const QueryParamName = {
```
Comment (bianucci) [inline iris-client-fe/src/constants/QueryParams.ts:1]: Suggestion: 
1. rename file to: url-query-params.ts
2. rename const to PaginationQueryParams

File: iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue [hunk 7]
```diff
@@ -29,22 +29,27 @@
       <v-card-title>Ereignisnachverfolgungen</v-card-title>
       <v-card-text>
         <v-text-field
-          v-model="tableData.search"
+          v-model="search"
           append-icon="mdi-magnify"
-          label="Search"
+          label="Suche (min. 2 Buchstaben)"
```
Comment (bianucci) [inline iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue:34]: Suggestion: `Suchbegriff (min. 2 Buchstaben)"

File: iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue [hunk 8]
```diff
@@ -29,22 +29,27 @@
...
-          :items-per-page="5"
+          :page="dataTableOptions.page"
+          :server-items-length="eventList.totalElements"
+          :headers="headers"
+          :items="eventList.content"
+          :items-per-page="dataTableOptions.itemsPerPage"
           class="elevation-1 mt-5 twolineTable"
-          :search="tableData.search"
+          :search="search"
+          :footer-props="{ 'items-per-page-options': [5, 10, 15] }"
```
Comment (bianucci) [inline iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue:48]: Wondering: `5, 10, 15` are specified requirements or is it more of a good guess?
Maybe `10, 20, 30, 50` would be more helpful to users?

File: iris-client-fe/src/views/index-tracking-list/index-tracking-list.view.vue [hunk 9]
```diff
@@ -94,79 +108,163 @@ type TableRow = {
...
+  );
+
   selectableStatus = {
     ..._omit(DataRequestStatus, ["Aborted"]),
-    All: null,
+    All: undefined,
   };
-  statusButtonSelected = Object.keys(this.selectableStatus).length - 1;
-  filterStatus(target: DataRequestStatus | null): void {
+  statusButtonSelected = this.statusFilter
```
Comment (bianucci) [inline iris-client-fe/src/views/index-tracking-list/index-tracking-list.view.vue:125]: Lets make this a computed attribute to improve readability.

File: iris-client-fe/src/views/index-tracking-list/index-tracking-list.view.vue [hunk 10]
```diff
@@ -33,19 +33,24 @@
       <v-card-title>Indexfallverfolgung</v-card-title>
       <v-card-text>
         <v-text-field
-          v-model="tableData.search"
+          v-model="search"
           append-icon="mdi-magnify"
-          label="Search"
+          label="Suche (min. 2 Buchstaben)"
```
Comment (bianucci) [inline iris-client-fe/src/views/index-tracking-list/index-tracking-list.view.vue:38]: Suggestion: rename to `Suchbegriff (...`

File: iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue [hunk 11]
```diff
@@ -123,60 +138,110 @@ type TableRow = {
...
+        return v === this.statusFilter;
+      })
+    : Object.keys(this.selectableStatus).length - 1;
+
+  dataTableOptions = {
+    currentPage: getPageFromRouteWithDefault(this.$route),
+    itemsPerPage: getPageSizeFromRouteWithDefault(this.$route),
+  };
+
+  headers = [
```
Comment (bianucci) [inline iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue:165]: I guess `headers` should be part of dataTableOptions.

```
dataTableOptions = {
  currentPage: getPageFromRouteWithDefault(this.$route),
  itemsPerPage: getPageSizeFromRouteWithDefault(this.$route),
  headers: [
````

No strong feelings though

**Review by bianucci — CHANGES_REQUESTED** [review]
see comments

File: iris-client-bff/src/main/java/iris/client_bff/cases/web/IndexCaseController.java (hunk 1 above)
Comment (faversente) [reply iris-client-bff/src/main/java/iris/client_bff/cases/web/IndexCaseController.java]: Done for `search`. `status` is an Enum

File: iris-client-bff/src/main/java/iris/client_bff/events/web/EventDataRequestController.java (hunk 2 above)
Comment (faversente) [reply iris-client-bff/src/main/java/iris/client_bff/events/web/EventDataRequestController.java]: Done for `search`. `status` is an Enum

File: iris-client-fe/src/constants/QueryParams.ts (hunk 6 above)
Comment (faversente) [reply iris-client-fe/src/constants/QueryParams.ts]: was not used anymore, deleted

File: iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue (hunk 8 above)
Comment (faversente) [reply iris-client-fe/src/views/event-tracking-list/event-tracking-list.view.vue]: `5, 10, 15, All` are the default values in `v-data-table`. I also think `10, 20, 30, 50` are probably more useful and have changed them accordingly

File: iris-client-fe/src/api/common.ts (hunk 4 above)
Comment (faversente) [reply iris-client-fe/src/api/common.ts:158]: This interface is currently only used within the event/index-case lists, and acts as a mapper from PageIndexCase/PageEvent => TableRow specified for table data visualization.

Maybe it is sufficient to move and rename it?

**Review by bianucci — APPROVED** [review]

**Pull request merged**